pkg/engine/lifecycle_test.go

```diff
@@ -3161,4 +3162,18 @@ func TestAliases(t *testing.T) {
 		},
 	}}, []deploy.StepOp{deploy.OpReplace, deploy.OpCreateReplacement, deploy.OpDeleteReplaced})
 
+	// Ensure that changing a forceNew property and deleteBeforeReplace while also changing type and name leads to
+	// replacement not delete+create
+	_ = updateProgramWithResource(snap, []Resource{{
```

Comment:
It would be good to extend this test s.t. there are dependent resources that also require deletion

Reply:
Done - though not sure if there was a specific scenario you had in mind here.